hw/xbox/xbox.c

```diff
@@ -476,6 +476,34 @@ static void machine_set_eeprom(Object *obj, const char *value,
     ms->eeprom = g_strdup(value);
 }
 
+static char *machine_get_avpack(Object *obj, Error **errp)
+{
+    XboxMachineState *ms = XBOX_MACHINE(obj);
+
+    return g_strdup(ms->avpack);
+}
+
+static void machine_set_avpack(Object *obj, const char *value,
+                               Error **errp)
+{
+    XboxMachineState *ms = XBOX_MACHINE(obj);
+
+    if (strcmp(value, "composite") != 0 &&
+        strcmp(value, "scart") != 0 &&
+        strcmp(value, "svideo") != 0 &&
+        strcmp(value, "vga") != 0 &&
+        strcmp(value, "rfu") != 0 &&
+        strcmp(value, "hdtv") != 0 &&
+        strcmp(value, "none") != 0)
```

Comment:
nit: Not a big fan of all the duplications in this code. It's easy to break stuff by just changing the strings in one function, or by forgetting to add one of the items.

Also having this list hardcoded here makes it harder to add a monitor command later (duplicating this list *again*).

There aren't many clean options I can think of myself. I'd *personally* probably abuse `smbus_avpack_to_reg` and have it return an `int`, where `-1` simply means: no valid mapping found. Then you can do the error handling based on that (as that is still unique to each caller).
There might be better options (using a second argument for `bool* valid` return-value etc.) so it could be helpful to look how other files in QEMU handle this.

Again: this is nitpicking to me (for now). Other maintainers might think differently.

Reply:
Improved. :slightly_smiling_face:  I looked up how qemu handles errors in other realize functions and did it the same way.